kasa/kasacamera.py

```diff
@@ -0,0 +1,97 @@
+"""Module for KasaCams (EC70)."""
+from typing import Any, Dict, Optional
+
+from .kasacamprotocol import KasaCamCredentials, KasaCamProtocol
+from .modules import PTZ, Cloud, Time
+from .smartdevice import DeviceType, SmartDevice, requires_update
+
+
+class KasaCam(SmartDevice):
+    """Representation of a TP-Link Kasa Camera.
+
+    To initialize, you have to await :func:`update()` at least once.
+    This will allow accessing the properties using the exposed properties.
+
+    All changes to the device are done using awaitable methods,
+    which will not change the cached values, but you must await :func:`update()`
+    separately.
+
+    Errors reported by the device are raised as
+    :class:`SmartDeviceExceptions <kasa.exceptions.SmartDeviceException>`, and should be
+    handled by the user of the library.
+
+    Examples:
+        >>> import asyncio
+        >>> camera = KasaCam("127.0.0.1")
+        >>> asyncio.run(camera.update())
+        >>> print(camera.alias)
+        Camera2
+
+        Cameras, like any other supported devices, can be turned on and off:
+
+        >>> asyncio.run(camera.turn_off())
+        >>> asyncio.run(camera.turn_on())
+        >>> asyncio.run(camera.update())
+        >>> print(camera.is_on)
+        True
+    """
+
+    def __init__(
+        self,
+        host: str,
+        credentials: KasaCamCredentials,
+        *,
+        port: Optional[int] = None,
+        timeout: Optional[int] = None,
+    ) -> None:
+        super().__init__(host, port=port, credentials=credentials, timeout=timeout)
+        self._device_type = DeviceType.KasaCam
+        self.protocol = KasaCamProtocol(
+            host, credentials=credentials, port=port, timeout=timeout
+        )
+        self.add_module("cloud", Cloud(self, "smartlife.cam.ipcamera.cloud"))
+        self.add_module("ptz", PTZ(self, "smartlife.cam.ipcamera.ptz"))
+        self.add_module("time", Time(self, "smartlife.cam.ipcamera.dateTime"))
+
+    def _create_request(
+        self,
+        target: str,
+        cmd: str,
+        arg: Optional[Dict] = {},  # noqa: B006 - Kasa Cam devices expect {} instead of None for empty args, all others accept it as well
+        child_ids=None,
+    ):
+        return {target: {cmd: arg}}
+
+    @property  # type: ignore
+    @requires_update
+    def is_on(self) -> bool:
+        """Return whether device is on."""
+        return self.sys_info["camera_switch"] == "on"
+
+    @property  # type: ignore
+    @requires_update
+    def state_information(self) -> Dict[str, Any]:
+        """Return camera-specific state information.
+
+        :return: Strip information dict, keys in user-presentable form.
+        """
+        return {
+            "Camera state": self.is_on,
+        }
+
+    @property  # type: ignore
+    def is_camera(self) -> bool:
+        """Return whether this device is a camera."""
+        return True
```

Comment:
This should be moved to `SmartDevice` and converted to use the device type like the rest of `is_X` properties.

Reply:
fixed